pkg/hintrunner/zero/zerohint_keccak.go

```diff
@@ -102,3 +105,93 @@ func createKeccakWriteArgsHinter(resolver hintReferenceResolver) (hinter.Hinter,
 
 	return newKeccakWriteArgsHint(inputs, low, high), nil
 }
+
+func newBlockPermutationHint(keccakStateSizeFelts, keccakPtr hinter.ResOperander) hinter.Hinter {
+	name := "BlockPermutation"
+	return &GenericZeroHinter{
+		Name: name,
+		Op: func(vm *VM.VirtualMachine, _ *hinter.HintRunnerContext) error {
+			// from starkware.cairo.common.keccak_utils.keccak_utils import keccak_func
+			// _keccak_state_size_felts = int(ids.KECCAK_STATE_SIZE_FELTS)
+			// assert 0 <= _keccak_state_size_felts < 100
+
+			// output_values = keccak_func(memory.get_range(
+			// 	ids.keccak_ptr - _keccak_state_size_felts, _keccak_state_size_felts))
+			// segments.write_arg(ids.keccak_ptr, output_values)
+
+			keccakStateSize, err := hinter.ResolveAsUint64(vm, keccakStateSizeFelts)
+			if err != nil {
+				return err
+			}
+
+			keccakWritePtr, err := hinter.ResolveAsAddress(vm, keccakPtr)
+			if err != nil {
+				return err
+			}
+
+			// why does the need contains this check? keccakStateSize is a constant: 25
+			if keccakStateSize >= 100 {
+				return fmt.Errorf("keccakStateSize %v is out range 0 <= keccakStateSize < 100", &keccakStateSize)
+			}
+
+			var readAddr = *keccakWritePtr
+			var offset int16 = int16(keccakStateSize)
+			var negOffset int16 = -offset
+
+			readAddr, err = readAddr.AddOffset(negOffset)
```

Comment:
Did you check what happens, when the address resulting in this subtraction is negative?

Reply:
i didnt actually. Will do it. I I still need to run the hint on the python vm and do tests